dask/dataframe/tests/test_multi.py

```diff
@@ -1386,14 +1389,19 @@ def test_cheap_single_partition_merge():
     b = pd.DataFrame({"x": [1, 2, 3, 4], "z": list("abda")})
     bb = dd.from_pandas(b, npartitions=1, sort=False)
 
-    cc = aa.merge(bb, on="x", how="inner")
+    pd_inputs = (b, a) if flip else (a, b)
+    inputs = (bb, aa) if flip else (aa, bb)
+
+    cc = dd.merge(*inputs, on="x", how="inner")
+    assert not hlg_layer_topological(cc.dask, -1).is_materialized()
     assert all("shuffle" not in k[0] for k in cc.dask)
     assert len(cc.dask) == len(aa.dask) * 2 + len(bb.dask)
 
-    list_eq(aa.merge(bb, on="x", how="inner"), a.merge(b, on="x", how="inner"))
+    list_eq(cc, pd.merge(*pd_inputs, on="x", how="inner"))
 
 
-def test_cheap_single_partition_merge_divisions():
+@pytest.mark.parametrize("flip", [False, True])
+def test_cheap_single_partition_merge_divisions(flip):
```

Comment:
It looks like we're not using `flip` in this test

Reply:
Good catch, thanks! The test already tried both orders, so the `flip` was unnecessary.